gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -312,8 +313,16 @@ protected GetFilePathsToFileStatusResult getFilePathsToFileStatus(FileSystem tar
       wrapper.rethrowWrapped();
     }
 
-    if (readTimeTableMetadataHolder.size() == 0) {
-      throw new RuntimeException(String.format("~%s~ no table metadata ever encountered!", this.getFileSetId()));
+    if (readTimeTableMetadataHolder.size() != 1) {
+      final int firstNumToShow = 5;
+      String errMsg = readTimeTableMetadataHolder.size() == 0
+          ? String.format("~%s~ no table metadata ever encountered!", this.getFileSetId())
+          : String.format("~%s~ multiple metadata (%d) encountered (exactly 1 expected) - first %d: [%s]",
+              this.getFileSetId(), readTimeTableMetadataHolder.size(), firstNumToShow,
+              readTimeTableMetadataHolder.stream().limit(firstNumToShow).map(md ->
+                  md.uuid() + " - " + md.location()
+              ).collect(Collectors.joining("\n", "\n", "\n")));
```

Comment:
Nit: `\n` should be `System.lineSeparator()`

Reply:
changed